crates/sncast/src/helpers/interactive.rs

```diff
@@ -0,0 +1,193 @@
+use crate::helpers::config::get_global_config_path;
+use anyhow::{anyhow, Context, Result};
+use camino::Utf8PathBuf;
+use configuration::search_config_upwards_relative_to;
+use dialoguer::theme::ColorfulTheme;
+use dialoguer::Select;
+use std::{env, fs};
+use toml_edit::{DocumentMut, Item, Table, Value};
+
+pub fn prompt_to_add_account_as_default(account: &str) -> Result<()> {
+    let mut options = Vec::new();
+
+    if let Some(option) = get_global_config_path().ok().map(|global_path| {
+        format!(
+            "Yes, global default account ({}).",
+            to_tilde_path(&global_path)
+        )
+    }) {
+        options.push(option);
+    }
+
+    if let Some(option) = env::current_dir()
+        .ok()
+        .and_then(|current_path| Utf8PathBuf::from_path_buf(current_path.clone()).ok())
+        .and_then(|current_path_utf8| search_config_upwards_relative_to(&current_path_utf8).ok())
+        .map(|local_path| {
+            format!(
+                "Yes, local default account ({}).",
+                to_tilde_path(&local_path)
+            )
+        })
+    {
+        options.push(option);
+    }
+
+    let selection = Select::with_theme(&ColorfulTheme::default())
+        .with_prompt("Do you want to make this account default?")
+        .items(&options)
+        .default(0)
+        .interact()
+        .context("Failed to display selection dialog")?;
+
+    match options[selection].as_str() {
+        selected if selected.starts_with("Yes, global default") => {
+            if let Ok(global_path) = get_global_config_path() {
+                edit_config(&global_path, "default", "account", account)?;
+            }
+        }
+        selected if selected.starts_with("Yes, local default") => {
+            if let Ok(current_path) = env::current_dir() {
+                let current_path_utf8 = Utf8PathBuf::from_path_buf(current_path).map_err(|_| {
+                    anyhow!("Failed to convert current directory path to Utf8PathBuf")
+                })?;
+
+                if let Ok(local_path) = search_config_upwards_relative_to(&current_path_utf8) {
```

Comment:
We're duplicating the search logic, can we store the local path if it exists and use it here and in the line 22?